### PR TITLE
Introduce and use the Date type

### LINKED ISSUE #2162: Date format
Some have pushed back on using integers for dates, see eg https://github.com/ietf-wg-httpapi/deprecation-header/issues/11

As I suggested there, one accommodation would be to define a new structured type for them. That could be done in retrofit, or separately.

### PR DESCRIPTION
**Pull request by mnot**
Fixes #2162.

### CODE REVIEW

File: draft-ietf-httpbis-retrofit.md [hunk 1]
```diff
@@ -291,14 +295,14 @@ Cookie attributes map to Parameters on the Inner List, with the parameter name b
...
-| Expires             | Integer             |
+| Expires             | Date                |
 | Max-Age             | Integer             |
 | Path                | String              |
 | Secure              | Boolean             |
 | SameSite            | Token               |
 {:id="cookie-params" title="Set-Cookie Parameter Types"}
 
-The Expires attribute is mapped to an Integer representation of parsed-cookie-date (see {{Section 5.1.1 of COOKIES}}) expressed as a number of seconds delta from the Unix Epoch (00:00:00 UTC on 1 January 1970, excluding leap seconds).
+The Expires attribute is mapped to an Date representation of parsed-cookie-date (see {{Section 5.1.1 of COOKIES}}).
```
Comment (MikeBishop) [inline draft-ietf-httpbis-retrofit.md:305]:
```suggestion
The Expires attribute is mapped to a Date representation of parsed-cookie-date (see {{Section 5.1.1 of COOKIES}}).
```

File: draft-ietf-httpbis-retrofit.md [hunk 2]
```diff
@@ -380,4 +384,39 @@ Finally, add a new column to the "Cookie Attribute Registry" established by {{CO
 --- back
 
 
+# The Date Structured Type {#date-type}
+
+This section defines a new Structured Fields data type, Date.
+
+Dates have a data model that is similar to Integers, representing a (possibly negative) delta in seconds from January 1, 1970 00:00:00 UTC, excluding leap seconds.
```
Comment (MikeBishop) [inline draft-ietf-httpbis-retrofit.md:391]: Should this definition (converting to an actual date) be embedded in the serializing/parsing instructions, such that the output is actually a date?  Or perhaps just a note that converting an integer to a date representation will be implementation-dependent.

**Review by MikeBishop — APPROVED** [review]
I like this direction; thanks.

File: draft-ietf-httpbis-retrofit.md (hunk 2 above)
Comment (mnot) [reply draft-ietf-httpbis-retrofit.md:391]: That would require us to define / choose a model for dates. I'd rather just stay with delta to epoch - it's widely used and understood.

**Comment by reschke** [comment]
I still believe this change should end up in the revision of SF.

**Comment by mnot** [comment]
That's #2225

**Pull request merged**